### ISSUE TITLE
Migrating from vue-class-component

### ISSUE THREAD

**Issue by nseb** [facing-dev/vue-facing-decorator#55]
Hello , 
with vue-class-component, it's works : 
@ComponentBase({
  name: 'UMBaseComponent'
})
export default class UMBaseComponent extends Vue {
  $q: QVueGlobals = useQuasar();
  quasar : QVueGlobals = useQuasar();
  ....

@Component({
  name: 'LoginAvatar',
  components: {Avatar, AccountPanel, Languages, Notifications, FollowedDriversPanel}
})
export default class LoginAvatar extends mixins(UMBaseComponent, AuthComponent) {
  darkMode = **this.quasar**.dark.isActive;
...

with vue-facing-decorator , there is an error ; this.quasar is undefined , I know I can change the code , but I would like understand.

**Comment by ruojianll** [comment]
Use `@Setup`
```ts
https://github.com/component({
name: 'LoginAvatar',
components: {Avatar, AccountPanel, Languages, Notifications, FollowedDriversPanel}
})
export default class LoginAvatar extends mixins(UMBaseComponent, AuthComponent) {
@Setup(useQuasar)
$q!: QVueGlobals
...
```

**Comment by ruojianll** [comment]
> @ruojianll thanks , but it's not possible to refer component parent during initialize child ?

@nseb  What's your mean?

**Comment by rdhelms** [comment]
@nseb Were you able to resolve this issue? If so, would you be able to share your solution?

**Comment by ruojianll** [comment]
> @ruojianll thanks , but it's not possible to refer component parent during initialize child ?

No, this context can't be accessed when declaring a class field. It will be totally removed in the next version.
